File: 2942. Find Words Containing Character.py
#2942. Find Words Containing Character
words = ["leet","code"]
x = "e"
#*MY SOL
class Solution:
    def findWordsContaining(self, words, x):
        res=[]
        for index, word in enumerate(words):
            if x in word:
                res.append(index)
        return res


s=Solution()
print(s.findWordsContaining(words,x))


#*LEARN
# Example: Square all even numbers from 1 to 10
squares_of_evens = [x**2 for x in range(1, 11) if x % 2 == 0]
print(squares_of_evens)


# numbers = [1, 2, 3, 4, 5]
# even_numbers = numbers.select { |x| x.even? }

# Python:
numbers = [1, 2, 3, 4, 5]
even_numbers = filter(lambda x: x % 2 == 0, numbers)

File: 2942. Find Words Containing Character.js
/**
 * @param {string[]} words
 * @param {character} x
 * @return {number[]}
 */

/**
 * @param {string[]} words
 * @param {character} x
 * @return {number[]}
 */
//SOL 1.2
var findWordsContaining = function(words, x) {
  return words.map((word, index) => word.includes(x) ? index : undefined).filter(index => index !== undefined);
};

//SOL 1.1
var findWordsContaining = function(words, x) {
  const includesx = words.map((word, index) => word.includes(x) ? index : undefined).filter(index => index !== undefined);
  return includesx
};

//SOL 1
var findWordsContaining = function(words, x) {
  const includesx = words.map((word, index) => {
  if (word.includes(x)) {
      return index;
  }
  }).filter(index => index !== undefined);
  return includesx
};

